packages/smooth_app/lib/pages/product/common/product_list_popup_items.dart

```diff
@@ -216,6 +222,112 @@ class ProductListPopupOpenInWeb extends ProductListPopupItem {
+  @override
+  String getTitle(final AppLocalizations appLocalizations) =>
+      appLocalizations.product_list_export;
+
+  @override
+  IconData getIconData() => Icons.download;
+
+  @override
+  ProductListPopupMenuEntry getEntry() => ProductListPopupMenuEntry.export;
+
+  @override
+  bool isDestructive() => false;
+
+  @override
+  Future<ProductList?> doSomething({
+    required final ProductList productList,
+    required final LocalDatabase localDatabase,
+    required final BuildContext context,
+  }) async {
+    final String listName = ProductQueryPageHelper.getProductListLabel(
+      productList,
+      AppLocalizations.of(context),
+    );
+    final String fileName =
+        "${listName.replaceAll(' ', '-').toLowerCase()}_${DateTime.now().millisecondsSinceEpoch}.csv";
+    final StringBuffer csv = StringBuffer('Barcode,Name,Brand\n');
+
+    for (final String barcode in productList.barcodes) {
+      final Product? product = await DaoProduct(localDatabase).get(barcode);
+      if (product != null) {
+        final String name = (product.productName ?? '').replaceAll(';', '');
+        final String brand = (product.brands ?? '').replaceAll(';', '');
+        csv.write('"$barcode";"$name";"$brand"\n');
+      }
+    }
+
+    unawaited(
+      Share.shareXFiles(
+        <XFile>[
+          XFile.fromData(
+            utf8.encode(csv.toString()),
+            name: '$fileName.csv',
+            mimeType: 'text/csv',
+          ),
+        ],
+        fileNameOverrides: <String>['$fileName.csv'],
+      ),
+    );
+
+    return null;
+  }
+}
+
+/// Popup menu item for the product list page: import list.
+class ProductListPopupImport extends ProductListPopupItem {
+  @override
+  String getTitle(final AppLocalizations appLocalizations) =>
+      appLocalizations.product_list_import;
+
+  @override
+  IconData getIconData() => Icons.upload;
+
+  @override
+  ProductListPopupMenuEntry getEntry() => ProductListPopupMenuEntry.import;
+
+  @override
+  bool isDestructive() => false;
+
+  @override
+  Future<ProductList?> doSomething({
+    required final ProductList productList,
+    required final LocalDatabase localDatabase,
+    required final BuildContext context,
+  }) async {
+    FilePicker.platform.pickFiles(
+      type: FileType.custom,
+      allowedExtensions: <String>['csv'],
+    ).then((final FilePickerResult? result) {
+      if (result == null) {
+        return;
+      }
+      final File file = File(result.files.single.path!);
+      final String content = file.readAsStringSync();
+      final List<String> lines = content.split('\n');
+      final List<String> barcodes = <String>[];
+      for (final String line in lines) {
+        final List<String> parts = line.split(';');
+        if (parts.length < 3) {
+          continue;
+        }
+        final String barcode = parts[0].replaceAll('"', '');
+        barcodes.add(barcode);
+      }
+      DaoProductList(localDatabase).bulkSet(
```

Comment:
This stinks: it's a `Future` and therefore deserves an `await`.

I guess we should enforce the [`unawaited_futures`](https://dart.dev/tools/linter-rules/unawaited_futures#:~:text=Future%20results%20in%20async%20function%20bodies%20must%20be,a%20Future%20inside%20of%20an%20async%20function%20body.) rule.
@g123k Do you agree with me on that?

Reply:
I've awaited the call. Leaving open for the rule question.

Reply:
> I've awaited the call. Leaving open for the rule question.

"rule question"?

Reply:
If the [unawaited_futures](https://dart.dev/tools/linter-rules/unawaited_futures#:~:text=Future%20results%20in%20async%20function%20bodies%20must%20be,a%20Future%20inside%20of%20an%20async%20function%20body.) rule should be enforced or not

Reply:
> If the [unawaited_futures](https://dart.dev/tools/linter-rules/unawaited_futures#:~:text=Future%20results%20in%20async%20function%20bodies%20must%20be,a%20Future%20inside%20of%20an%20async%20function%20body.) rule should be enforced or not

Oh! I meant that generally it would be a good idea to enforce the `unawaited_futures` rule so that our code is cleaner: when we call a `Future` without `await`ing, we have to put `unawaited`, it's not something we forgot, it's a choice (like in your "share" code).
In that specific case it's a no brainer: you need to `await` in order to `return` something relevant.